control-plane/stor-port/src/types/v0/transport/snapshot.rs

```diff
@@ -0,0 +1,18 @@
+use super::*;
+use crate::transport_api::ReplyError;
+use async_trait::async_trait;
+use serde::Serialize;
+
+rpc_impl_string_uuid!(SnapshotId, "UUID of a snapshot");
+
+/// The trait for snapshot operations like create, remove, list.
+#[async_trait]
+pub(crate) trait SnapshotApi {
```

Comment:
hmm I don't think these trait should be here, but rather on the controller/io_engine...

Reply:
If I move this to controller/io-engine/mod.rs , it won't allow to use/import SnapshotApi in the transport/nexus.rs I think. Is that possible?

Reply:
No, that's a point, these traits are just for the core-agent controller io-engine.
The transport traits will come later as part of another PR related to grpc transport for the control-plane.